common/Util.cpp

```diff
@@ -1042,6 +1043,52 @@ namespace Util
 
         return StringVector(s, std::move(tokens));
     }
+
+    int safe_atoi(const char* p, int len)
+    {
+        long ret{};
+
+        int multiplier = 1;
+        int offset = 0;
+        while (isspace(p[offset]))
+        {
+            ++offset;
+            if (offset >= len)
+            {
+                return ret;
+            }
+        }
+
+        switch (p[offset])
+        {
+            case '-':
+                multiplier = -1;
+                ++offset;
+                break;
+            case '+':
+                ++offset;
+                break;
+        }
+        if (offset >= len)
+        {
+            return multiplier * ret;
+        }
+
+        while (isdigit(p[offset]))
+        {
+            long next = ret * 10 + (p[offset] - '0');
+            if (next >= std::numeric_limits<int>::max())
```

Comment:
This relies on long being longer than int - so this would fail in a hypothetical Windows port.

A safer would be to check if ret is greater than (max / 10 - newDigit) before addition.

Reply:
Hmm, but you need to store the "next" value if you don't want to calc it twice. :-) So I just used an explicit std::int64_t here instead of long.

Reply:
Sorry, missed the response - thanks, of course explicit 64-bit value works fine. Just FTR, I didn't suggest to calculate 'next' twice - I was thinking about something like

        while (isdigit(p[offset]))
        {
            int digit = p[offset] - '0';
            if (ret >= std::numeric_limits<int>::max() / 10 - digit) // this only does subtraction, since std::numeric_limits<int>::max() / 10 is compile-time constant
                return multiplier * std::numeric_limits<int>::max();
            std::int64_t next = ret * 10 + digit;
            ...